rust/protocol/src/uuid.rs

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright 2021 Signal Messenger, LLC.
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+use std::array::TryFromSliceError;
+use std::convert::TryFrom;
+
+#[derive(PartialEq, Eq, Hash, Clone, Copy, Debug)]
+pub struct Uuid([u8; 16]);
```

Comment:
This is very similar to the [`uuid` crate](https://docs.rs/uuid/0.8.2/uuid/), but I wasn't sure if we wanted to pull in another dependency.

Reply:
This seems simple enough to just use inline without adding a dep, IMO